### ISSUE TITLE
Build and deploy web app and api as separate actions

### ISSUE THREAD

**Issue by simensma-fresh** [bcgov/digital-journeys#723]
Currently, the web app and api are built using the same Github action. This makes the pipelines pretty slow when you only want to deploy one of them as you have to wait for one to finish before the other one can build.

What we want to accomplish is to build the web app and api separately, and move the actual build step into the github actions themselves (out of openshift) so the builds and deployments can happen in parallel.

*Acceptance Criteria*
- When a PR is merged into the dev branch, builds of the web app and web api are triggered as separate github actions
- The web api build is only triggered when changes have been made in the forms-flow-api folder
- The web app build is only triggered when changes have been made in the forms-flow-web folder
- I can trigger a build of the web app to dev, test and production from the "actions" tab
- I can trigger a build of the web api to dev, test and production from the "actions" tab 
- Builds of the web app / web api should happen in the github pipeline (build-push-action) instead of using OpenShift build configs 

*Suggested Approach*

- Modify `workflows/forms-flow-web-cd.yml` to automatically run on pushes to the `dev` branch, and change the conditional logic there that checks whether or not its the develop vs master branch to instead check the environment to deploy to based on the `workflow_dispatch` input (example in `workflows/deploy.yaml`). 
- Push the built image to the internal openshift docker registry, you can do that in the [docker/login-action](https://github.com/docker/login-action), just specify the url to the registry
- Docker repository is available at: image-registry.apps.silver.devops.gov.bc.ca/d89793-tools, you can find credentials in the d89793-tools project: User Management -> Service Accounts -> github-actions-sa -> deployer-token. You can test it out by doing `docker login image-registry.apps.silver.devops.gov.bc.ca/d89793-tools` and then `docker images`

**Comment by bhumin-fw** [comment]
@simensma-fresh 
I'm not able to login into PCoD, so not able to check mail or teams. Could you share meeting link on my email bhumin.patel@freshworks.io or here in this ticket? If you got some time to walk us through openshift dev.
: I'll delete this message once PCoD start working :)
